Ticket/aotech 6705 date based deltas

phatsk opened the pull request:
This change allows the user to set a date by which posts modified after are the only posts synced.

phatsk committed:
update options for advanced export.

phatsk committed:
Add ability to do a post-sync using a date for determining deltas.

jmichaelward commented on includes/class-dashboard.php:
```diff
@@ -141,14 +141,14 @@ public function register_settings() {
 		// @TODO update option names and locations below this line.
 		// Export page.
 		register_setting( 'press-sync-export', 'ps_options' );
-		register_setting( 'press-sync-export', 'ps_ignore_comments' );
```
Why is this removed as part of this ticket?

phatsk replied:
It could easily be moved to its separate issue - I just noticed that that was where a different small bug was coming from. I'll remove it from this PR.

jmichaelward commented on includes/class-press-sync.php:
```diff
@@ -414,8 +424,9 @@ public function count_objects_to_sync( $objects_to_sync ) {
-		$where_clause = '';
-		$where_clause = $this->get_synced_object_clause( $objects_to_sync );
+		$where_clause  = '';
+		$where_clause  = $this->get_synced_object_clause( $objects_to_sync );
```
This line just overwrites the previous line. Is the previous line needed?

phatsk replied:
tbh I wasn't really paying attention to the lines here and above, it's just marked "changed" from indentation. Will note that for the future, however.

jmichaelward commented on views/dashboard/html-export.php:
```diff
@@ -68,4 +74,8 @@
 	</form>
 </div>
-
+<script>
```
Nitpick: This is probably fine for now, but if we're on a custom admin screen we should really be enqueuing this JavaScript to load on that screen and not assigning it inline with our markup.

phatsk replied:
Totally agree - actually just forgot to put it in the proper place, will update.

rakeshdx commented on views/dashboard/html-export.php:
```diff
@@ -38,6 +38,12 @@
 				</td>
 			</tr>
+			<tr valign="top">
+				<th scope="row">Sync Post Since Date</th>
```
We should probably specify the datetime as "EST" since we dont use unix timestamp or GMT time.

phatsk replied:
Good call, will add that.

jmichaelward commented on includes/class-press-sync.php:
```diff
@@ -46,6 +54,7 @@ class Press_Sync {
 	public function __construct() {
 		// Initialize plugin classes.
 		$this->plugin_classes();
+		$this->delta_date = date( 'Y-m-d 00:00:00', strtotime( get_option( 'ps_delta_date' ) ) ) ?: false;
```
If it's already in the database we probably have bigger problems, but we're not sanitizing the ps_delta_date value here before we're using it and passing it to the date method. It'll probably make your date() method flunk out of class long before it gets appended as WHERE condition in your future SQL, but it's worth double-checking this to make sure there are no security implications here.

phatsk replied:
Considering `date` returns `false` on any non-numeric data, I don't think it's a huge concern, however, it's enough that I agree that sanitization for sanity's sake isn't a bad idea. I'll add that in.

rakeshdx commented on includes/class-press-sync.php:
```diff
@@ -1529,4 +1540,25 @@ public function maybe_get_terms_for_post( $where ) {
+		}
+
+		// Currently only supported for posts.
+		if ( 'post' !== $objects_to_sync ) {
```
post meaning all different public post types? or just "post" post type

phatsk replied:
All different Custom Post Types etc (including Pages).

jmichaelward reviewed (changes requested):
It's probably okay, but there are some direct database value usage issues I raised that need a second look, and a couple of other minor nitpicks.

phatsk committed:
Updates from PR suggestions.

Restore bad config save for it's own task.
Use WPDB prepare for date comparison.
Add note that time is EST-based for date delta.

phatsk committed:
Merge branch 'master' into ticket/AOTECH-6705-date-based-deltas
